src/engine/core/ComponentMap.ts

```diff
@@ -92,11 +92,11 @@ export default class ComponentMap<T extends Component> {
    * Retrieves all entities with this component type.
    * @returns a cloned array of entities.
    */
-  getEntities(): Entity[] {
+  getEntities(): Set<Entity> {
     if (!this.cachedEntities) {
-      this.cachedEntities = Array.from(this.components.keys());
+      this.cachedEntities = new Set(this.components.keys());
     }
-    return [...this.cachedEntities];
+    return this.cachedEntities;
```

Comment:
Anledningen till [... this.cachedEntities] är för att det ska skapas en kopia så att inte användaren kan ändra interna datan. Det kan användaren nu. Men kanske itne är ett problem?

Man kan skapa ett nytt set innan man returnerar om man vill?